### ISSUE TITLE
Parse CUDA Library constants from .elf file

### ISSUE THREAD

**Issue by drichmond** [bespoke-silicon-group/bsg_f1#175]
https://github.com/bespoke-silicon-group/bsg_f1/blob/b27a138286b1840dee57de4086bbf5fcfc0d9cea/cl_manycore/libraries/bsg_manycore_cuda.cpp#L15-L21

Is it possible to pull these constants from symbols in the elf file?

**Comment by drichmond** [comment]
DMEM_BASE should also be parsed from there (I think)

**Comment by drichmond** [comment]
Is this stale? (or will be with @bornaehsani's changes?)

**Comment by bornaehsani** [comment]
Not stale, but it is has priority on my todo list. Will try to address it soon.

**Comment by max-ruttenberg** [comment]
This looks like you could do this with symbol_to_eva?

**Comment by bornaehsani** [comment]
Yeah the problem is not on the host side, I can use symbol to eva IF we had a symbol to begin with, problem is I can't define them as a symbols and use them in the riscv assembly, the assembly doesn't take symbols (or at least I couldn't find a way to do it).

**Comment by vb000** [comment]
@bornaehsani Could you refer `crt.S` in bsg_manycore? It uses symbols in assembly...

**Comment by max-ruttenberg** [comment]
@bornaehsani checkout the input/output operands https://gcc.gnu.org/onlinedocs/gcc/Extended-Asm.html

Here's an example.
```c

uint32_t r_global = 0xDEADBEEF;
uint32_t w_global;

int main()
{       
        uint32_t r, w = 0xCAFEBABE;

        // r = r_global
        // w_global = w
        asm ("lw   %0, %2\n"
             "sw   %3, %1\n"
             : "=r"(r), "=m"(w_global)
             : "m"(r_global), "r" (w)
             : "memory" );
               
        bsg_printf("r=%u, w_global = %u\n", r, w_global);
               
}
```

**Comment by bornaehsani** [comment]
wow this looks great! I'll try to implement it, thank you both!

**Comment by bornaehsani** [comment]
Addressed in PR #298 .
